video/out/vo_opengl_cb.c

```diff
@@ -479,7 +484,13 @@ static int reconfig(struct vo *vo, struct mp_image_params *params)
 
 void mp_client_set_icc_profile(struct mpv_opengl_cb_context *ctx, bstr icc_data)
 {
+    pthread_mutex_lock(&ctx->icc_lock);
```

Comment:
Would it be possible to always call the function on the renderer thread? That's probably also how an official opengl-cb API for this would work.

Reply:
oh do you mean like it was [before](https://github.com/mpv-player/mpv/pull/4833/commits/ad89cb84b59216fbf1906b41972b67e211ce34fc#diff-b5010a29091b72c707c97eae67a97254L80) (thought the current approach was preferable) or do it in the mpv_opengl_cb_draw function via a flag?

Reply:
No, just rule that `mp_client_set_icc_profile` can be called in the GL thread only, so that no additional synchronization is needed.

Reply:
sure can do that. something broke anyway when i rebased it with the current "vo_gpu" changes. just getting a "blue screen" and shader errors on some icc profile changes.

should i mention that rule somewhere?

Reply:
As a comment on the function would be good.